source/platforms/git/gitJSONToGitDSL.ts

```diff
@@ -35,7 +36,7 @@ export interface GitJSONToGitDSLConfig {
   /** A promise which will return the string content of a file at a sha */
   getFileContents: (path: string, repo: string | undefined, sha: string) => Promise<string>
   /** A promise which will return the diff string content for a file between shas */
-  getFullDiff?: (base: string, head: string) => Promise<string>
+  getFullDiff?: (base: string, head: string, staging: boolean) => Promise<string>
```

Comment:
```suggestion
  getFullDiff?: (base: string, head: string, staging?: boolean) => Promise<string>
```

Is probably what you're looking for IMO

Reply:
:+1: thx!